onnx/backend/test/case/node/pool_op_common.py

```diff
@@ -27,10 +27,52 @@ def get_pad_shape(
         pass
     return pad_shape
 
+# Use pads to calculate output shape. Use output shape in turn to calculate the actual pads
+# that are used to pad the input tensor so that computation in pool() will not cause out of bound error.
+def get_output_shape_update_pads(
+    pads_spatial_shape: Sequence[int] | None,
+    input_spatial_shape: Sequence[int],
+    dialations_spatial_shape: Sequence[int],  # dialations_shape to be used
+    kernel_spatial_shape: Sequence[int],
+    strides_spatial: Sequence[int],
+    ceil_mode: bool,
+) -> Tuple[Sequence[int], Sequence[int]]:
+    
+    # compute output shape according to:
+    # https://pytorch.org/docs/stable/generated/torch.nn.MaxPool1d.html?highlight=max+pool#torch.nn.MaxPool1d
+    output_spatial_shape = [0] * len(input_spatial_shape)
+    dims = len(input_spatial_shape)
+    for dim in range(dims):
+        dim_size = (float)(input_spatial_shape[dim] + pads_spatial_shape[dim] + pads_spatial_shape[dims + dim] - dialations_spatial_shape[dim] * (kernel_spatial_shape[dim] - 1) - 1) / strides_spatial[dim] + 1
+        if ceil_mode:
+            output_spatial_shape[dim] = int(np.ceil(dim_size))
+        else:
+            output_spatial_shape[dim] = int(np.floor(dim_size))
+
+    # import numpy as np
```

Comment:
If it is important, you should put that in the docstring as well. It will show up in the documentation.